samples/sensor/tmp108/src/main.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2021 Jimmy Johnson <catch22@fastmail.net>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+
+#include <zephyr.h>
+#include <drivers/sensor.h>
+#include <drivers/sensor/tmp108.h>
+
+/* One shot requires a callback to be invoked after the tmp108 has
+ * woken up and taken a temperature reading, calling sample_fetch
+ * starts this process. Set this to 1 to enable 1 shot mode example.
+ */
+#define ENABLE_ONE_SHOT             0
+
+/* Set this to 1 to enable alerts for high and low temperatures
+ * although this will work with one shot enabled, it requires
+ * continuous monitoring mode to be enabled to work in real time.
+ * See spec sheet for more details.
+ */
+#define ENABLE_TEMP_ALERT           0
```

Comment:
use Kconfig for this, e.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/sensor/sgp40_sht4x/Kconfig